egs/wsj/s5/steps/nnet3/convert_nnet2_to_nnet3.py

```diff
@@ -0,0 +1,441 @@
+        pairs = ['{0}={1}'.format(TokenToString(key),pairs[key]) for key in pairs]
+        
+        # keep track of layer type number (e.g. affine3)
+        node_name = NODE_NAMES[component]
+        self.counts[node_name] += 1
+
+        # e.g. affine3
+        ident = node_name + str(self.counts[node_name])
+
+        # <PnormComponent> -> PnormComponent
+        component = component[1:-1]
+
+        self.components.append(Component(ident, component, pairs))
+
+    def WriteConfig(self, filename):
+        '''
+        Write config to filename.
+        '''
+        logger.info('Writing config to {0}'.format(filename))
+
+        self.config = filename
+        with open(filename, 'w') as f:
+            for component in self.components:
+                if component.ident == "splice":
+                    continue
+                config_string = ' '.join(component.pairs)
+
+                f.write('component name={name} type={comp_type} {config_string}\n'.format(name=component.ident, comp_type=component.component, config_string=config_string))
+
+            f.write('\n# Component nodes\n')
+            f.write('input-node name=input dim={0}\n'.format(self.input_dim))
+            previous_component='input'
+            for component in self.components:
+                if component.ident == "splice":
+                    # Create splice string for the next node
+                    previous_component=MakeSpliceString(previous_component, component.pairs['<Context>'])
+                    continue
+                f.write('component-node name={name} component={name} input={inp}\n'.format(name=component.ident, inp=previous_component))
+                previous_component = component.ident
+            logger.warning('Assuming linear objective.')
+            f.write('output-node name=output input={inp} objective={obj}\n'.format(inp=previous_component, obj='linear'))
+
+    def WriteModel(self, model, binary="true"):
+        result = 0
+
+        # write raw model
+        result += subprocess.call('nnet3-init --binary=true {0} {1}'.format(self.config, os.path.join(tmpdir, 'nnet3.raw')), shell=True)
+
+        # add transition model
+        result += subprocess.call('nnet3-am-init --binary=true {0} {1} {2}'.format(self.transition_model,
+                                                                                    os.path.join(tmpdir, 'nnet3.raw'),
+                                                                                    os.path.join(tmpdir, 'nnet3_no_prior.mdl')), shell=True)
+
+        # add priors
+        result += subprocess.call('nnet3-am-adjust-priors --binary={0} {1} {2} {3}'.format(binary, os.path.join(tmpdir, 'nnet3_no_prior.mdl'), 
+                                                                                           self.priors, 
+                                                                                           model), shell=True)
+
+        if (result != 0):
```

Comment:
Do you think it might be worth adding a check in a top level shell script which performs forward prop through the nnet2 and nnet3 models and asserts that the values are within acceptable threshold ?